src/bench/mempool_eviction.cpp

```diff
@@ -21,116 +21,96 @@ static void AddTx(const CTransactionRef& tx, const CAmount& nFee, CTxMemPool& po
-    tx4.vin[1].prevout.SetNull();
-    tx4.vin[1].scriptSig = CScript() << OP_4;
-    tx4.vin[1].scriptWitness.stack.push_back({4});
-    tx4.vout.resize(2);
-    tx4.vout[0].scriptPubKey = CScript() << OP_4 << OP_EQUAL;
-    tx4.vout[0].nValue = 10 * COIN;
-    tx4.vout[1].scriptPubKey = CScript() << OP_4 << OP_EQUAL;
-    tx4.vout[1].nValue = 10 * COIN;
-
-    CMutableTransaction tx5 = CMutableTransaction();
-    tx5.vin.resize(2);
-    tx5.vin[0].prevout = COutPoint(tx4.GetHash(), 0);
-    tx5.vin[0].scriptSig = CScript() << OP_4;
-    tx5.vin[0].scriptWitness.stack.push_back({4});
-    tx5.vin[1].prevout.SetNull();
-    tx5.vin[1].scriptSig = CScript() << OP_5;
-    tx5.vin[1].scriptWitness.stack.push_back({5});
-    tx5.vout.resize(2);
-    tx5.vout[0].scriptPubKey = CScript() << OP_5 << OP_EQUAL;
-    tx5.vout[0].nValue = 10 * COIN;
-    tx5.vout[1].scriptPubKey = CScript() << OP_5 << OP_EQUAL;
-    tx5.vout[1].nValue = 10 * COIN;
-
-    CMutableTransaction tx6 = CMutableTransaction();
-    tx6.vin.resize(2);
-    tx6.vin[0].prevout = COutPoint(tx4.GetHash(), 1);
-    tx6.vin[0].scriptSig = CScript() << OP_4;
-    tx6.vin[0].scriptWitness.stack.push_back({4});
-    tx6.vin[1].prevout.SetNull();
-    tx6.vin[1].scriptSig = CScript() << OP_6;
-    tx6.vin[1].scriptWitness.stack.push_back({6});
-    tx6.vout.resize(2);
-    tx6.vout[0].scriptPubKey = CScript() << OP_6 << OP_EQUAL;
-    tx6.vout[0].nValue = 10 * COIN;
-    tx6.vout[1].scriptPubKey = CScript() << OP_6 << OP_EQUAL;
-    tx6.vout[1].nValue = 10 * COIN;
-
-    CMutableTransaction tx7 = CMutableTransaction();
-    tx7.vin.resize(2);
-    tx7.vin[0].prevout = COutPoint(tx5.GetHash(), 0);
-    tx7.vin[0].scriptSig = CScript() << OP_5;
-    tx7.vin[0].scriptWitness.stack.push_back({5});
-    tx7.vin[1].prevout = COutPoint(tx6.GetHash(), 0);
-    tx7.vin[1].scriptSig = CScript() << OP_6;
-    tx7.vin[1].scriptWitness.stack.push_back({6});
-    tx7.vout.resize(2);
-    tx7.vout[0].scriptPubKey = CScript() << OP_7 << OP_EQUAL;
-    tx7.vout[0].nValue = 10 * COIN;
-    tx7.vout[1].scriptPubKey = CScript() << OP_7 << OP_EQUAL;
-    tx7.vout[1].nValue = 10 * COIN;
+    std::vector<CTransactionRef> txns;
+
+    for (size_t i=0; i<num_packages; i++) {
+        add_parents_child(txns, package_size, i);
+    }
 
+    /* Now we're ready */
     CTxMemPool& pool = *Assert(testing_setup->m_node.mempool);
     LOCK2(cs_main, pool.cs);
-    // Create transaction references outside the "hot loop"
-    const CTransactionRef tx1_r{MakeTransactionRef(tx1)};
-    const CTransactionRef tx2_r{MakeTransactionRef(tx2)};
-    const CTransactionRef tx3_r{MakeTransactionRef(tx3)};
-    const CTransactionRef tx4_r{MakeTransactionRef(tx4)};
-    const CTransactionRef tx5_r{MakeTransactionRef(tx5)};
-    const CTransactionRef tx6_r{MakeTransactionRef(tx6)};
-    const CTransactionRef tx7_r{MakeTransactionRef(tx7)};
+    bench.run([&]() NO_THREAD_SAFETY_ANALYSIS {
+        for (size_t i=0; i<txns.size(); i++) {
```

Comment:
nit: I think we tend to prefer this
```suggestion
        for (size_t i{0}; i < txns.size(); ++i) {
```